include/boost/geometry/algorithms/detail/closest_points/linear_to_linear.hpp

```diff
@@ -0,0 +1,155 @@
+// Boost.Geometry
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_LINEAR_TO_LINEAR_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_LINEAR_TO_LINEAR_HPP
+
+#include <boost/geometry/algorithms/detail/closest_points/range_to_geometry_rtree.hpp>
+#include <boost/geometry/algorithms/detail/closest_points/utilities.hpp>
+
+#include <boost/geometry/algorithms/num_points.hpp>
+#include <boost/geometry/algorithms/num_segments.hpp>
+
+#include <boost/geometry/core/point_type.hpp>
+
+#include <boost/geometry/iterators/point_iterator.hpp>
+#include <boost/geometry/iterators/segment_iterator.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace closest_points
+{
+
+
+struct linear_to_linear
+{
+    template <typename Linear1, typename Linear2, typename Segment, typename Strategies>
+    static inline void apply(Linear1 const& linear1,
+                             Linear2 const& linear2,
+                             Segment& shortest_seg,                       
+                             Strategies const& strategies,
+                             bool = false)
+    {
+        if (geometry::num_points(linear1) == 1)
+        {
+            dispatch::closest_points
+                <
+                    typename point_type<Linear1>::type,
+                    Linear2
+                >::apply(*points_begin(linear1), linear2, shortest_seg, strategies);
+            return;
+        }
+
+        if (geometry::num_points(linear2) == 1)
+        {
+            dispatch::closest_points
+                <
+                    typename point_type<Linear2>::type,
+                    Linear1
+                >::apply(*points_begin(linear2), linear1, shortest_seg, strategies);
+            detail::closest_points::swap_segment_points::apply(shortest_seg);
+            return;
+        }
+
+        if (geometry::num_segments(linear1) < geometry::num_segments(linear2))
+        {
+            point_or_segment_range_to_geometry_rtree::apply(
```

Comment:
This can be called for empty geometry which will cause assertion failure inside `point_or_segment_range_to_geometry_rtree`. Or am I missing something?

Reply:
Empty input is checked by the interface here: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/closest_points/interface.hpp#L178
Several empty input combinations are also tested in unit tests.